### PR TITLE
supporting Import CRD in policyengine

### LINKED ISSUE #331: Add load-balancing support to imports in CRD-mode
controlplane/authz should handle the loadbalancingspec and route accordingly.

### LINKED ISSUE #335: PolicyDispatcher should support Add{Import/Export} with the from the new pkg/apis

### PR DESCRIPTION
**Pull request by zivnevo**
In this PR:
* `AddImport` and `DeleteImport` now work with the new Import CRD
* Load-balancing scheme is now set by the Import CRD
* With static policy - the first service source is the default
* Support for namespaces

Fixes #331
Fixes #335

### CODE REVIEW

File: pkg/apis/clusterlink.net/v1alpha1/import.go [hunk 1]
```diff
@@ -50,7 +50,9 @@ type ImportSpec struct {
 	TargetPort uint16 `json:"targetPort,omitempty"`
 	// Sources to import from.
 	Sources []ImportSource `json:"sources"`
-	// TODO: add LoadBalancingSpec.
+	// LBScheme is the load-balancing scheme to use (e.g., Random, Static, ECMP)
+	LBScheme string `json:"lbScheme"`
+	// TODO: Make LBScheme a proper type (when backwards compatibility is no longer needed)
```
Comment (orozery) [inline pkg/apis/clusterlink.net/v1alpha1/import.go:55]: I don't see why moving the scheme definitions from `loadBalancer.go` into here will break anything.
But I can do it in a follow-up PR if you prefer.

File: pkg/policyengine/loadBalancer.go [hunk 2]
```diff
@@ -30,233 +33,180 @@ const (
...
+	impSources       []crds.ImportSource
 	totalConnections int
-	defaultPeer      string
 }
 
 type LoadBalancer struct {
-	ServiceMap      map[string][]string                 // Service to Peers
-	Scheme          map[string]map[string]LBScheme      // PolicyMap [serviceDst][serviceSrc]Policy
-	ServiceStateMap map[string]map[string]*ServiceState // State of policy Per destination and source
+	services map[string]*serviceState // Keeps a state for each imported service
```
Comment (orozery) [inline pkg/policyengine/loadBalancer.go:51]: Why not change the key from `string` to `types.NamespacedName` and avoid the string concatenations?

File: pkg/policyengine/loadBalancer.go [hunk 3]
```diff
@@ -30,233 +33,180 @@ const (
...
+func (lb *LoadBalancer) DeleteImport(impName types.NamespacedName) {
+	delete(lb.services, impName.String())
 }
 
-func (lB *LoadBalancer) removePeerFromService(svc, peer string) {
-	if peers, ok := lB.ServiceMap[svc]; ok {
-		index, exist := exists(peers, peer)
-		if !exist {
-			return
+// AddToServiceMap is being used by the CRUD interface and is now deprecated.
```
Comment (orozery) [inline pkg/policyengine/loadBalancer.go:82]: Who calls this?

File: pkg/policyengine/loadBalancer.go [hunk 4]
```diff
@@ -30,233 +33,180 @@ const (
...
-	}
 	return nil
 }
 
-func (lB *LoadBalancer) RemoveDestService(serviceDst, peer string) {
-	if peer != "" {
-		lB.removePeerFromService(serviceDst, peer)
-	} else {
-		delete(lB.ServiceMap, serviceDst)
+// RemoveDestService is being used by the CRUD interface and is now deprecated.
```
Comment (orozery) [inline pkg/policyengine/loadBalancer.go:125]: Same as `AddToServiceMap`

File: pkg/apis/clusterlink.net/v1alpha1/import.go (hunk 1 above)
Comment (zivnevo) [reply pkg/apis/clusterlink.net/v1alpha1/import.go:55]: Yes, I believe a separate PR is more proper

File: pkg/policyengine/loadBalancer.go (hunk 2 above)
Comment (zivnevo) [reply pkg/policyengine/loadBalancer.go]: Done

File: pkg/policyengine/loadBalancer.go (hunk 4 above)
Comment (zivnevo) [reply pkg/policyengine/loadBalancer.go]: Removed (dead code)

**Review by orozery — APPROVED** [review]

**Pull request merged**